keras_cv/models/object_detection/yolox/__internal__/layers/yolox_decoder.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+from tensorflow import keras
+
+import keras_cv.layers as cv_layers
+from keras_cv import bounding_box
+
+
+class DecodePredictions(keras.layers.Layer):
+    """Decodes the predictions from YoloX head.
+
+    This layer is similar to the decoding code in `YoloX.compute_losses`. This is
+    followed by a bounding box suppression layer.
+
+    Arguments:
+        bounding_box_format:  The format of bounding boxes of input dataset. Refer
+            [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+            for more details on supported bounding box formats.
+        classes: The number of classes to be considered for the classification head.
+            Defaults to None.
+        suppression_layer: A `keras.layers.Layer` that follows the same API
+            signature of the `keras_cv.layers.MultiClassNonMaxSuppression` layer.
+            This layer should perform a suppression operation such as Non Max Suppression,
+            or Soft Non-Max Suppression.
+    """
+
+    def __init__(self, bounding_box_format, classes, suppression_layer=None, **kwargs):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.classes = classes
+
+        self.suppression_layer = (
+            suppression_layer
+            or cv_layers.MultiClassNonMaxSuppression(
+                bounding_box_format=bounding_box_format,
+                from_logits=False,
+                confidence_threshold=0.01,
```

Comment:
@LukeWood I will definitely add comments to the code but addressing the 

> "YOLOx makes high confidence predictions, so we set a high confidence threshold"

 example here (because it was also mentioned in [this comment](https://github.com/keras-team/keras-cv/pull/1318#discussion_r1087413060) from #1318):

The default confidence threshold is quite low for YoloX (0.01; correct me if I understood your example wrong). Also, NMS is not a part of the training step at all. In fact, there's an NMS-free version proposed in the paper that performs slightly worse. 

I don't think that these values should be taken into consideration to make changes to the API btw, at least till I finalize numerical results (I might update it in the 3rd PR when I run numerical tests).

Reply:
sounds good to me - thanks for the explanation. 

Interesting these is a no-NMS version!  Curious to see how that works - I'll check it out.